### DISCUSSION TITLE
Mexico timezones wrong offsets

### DISCUSSION THREAD

**Discussion by RicardoM-Bonum** [moment/luxon#1432] [Q&A]
I have an application that uses time zones, the dates are saved in the database in UTC, when I display them on the client (a react application) they display correctly, in the case of the IANA America/Mexico_City zone, on the client It shows me an offset of -06:00 which is correct, but in the backend that same time zone shows me an offset of -05:00.

Does anyone have an idea?

**Comment by arcataroger** [comment]
What do you mean by:
> on the client It shows me an offset of -06:00 which is correct, but in the backend that same time zone shows me an offset of -05:00.

If the "dates are saved in the database in UTC", what does it mean when the "backend shows you an offset"? Like is it fetching from the database but then sending you an ISO string, or...?

In general, if you store a timestamp with an offset (like `2023-01-01T18:00:00-06:00` or  `2023-01-01T19:00:00-05:00` or `2023-01-02T00:00:00Z`, which are all the same time) in the backend, you should just be able to fetch that and then display it in whatever time zone you want. All those timestamps are the same as storing milliseconds directly. Then you can (and should) store the IANA string `America/Mexico_City` separately, fetch that _along with_ the timestamp, and then use that IANA zone to format the user-facing display appropriately (which is purely a frontend concern).

On the backend, a timestamp is a timestamp regardless of which zone it originated in. But on the frontend, an offset isn't enough to figure out the original timezone (because multiple time zones can share the same offset, but each time zone would have its own daylight savings rules or not, which can change over time). But if you store the timestamp+offset (or milliseconds) and also have the correct zone, the frontend can display that correctly.

Does that help at all, or am I way off the mark...?

**Reply by ricardoMogg** [reply]
Yes this is most likely a bug in luxon due to Mexico no longer using day light savings since last year.

**Reply by diesieben07** [reply]
Please see my comment in #1497: https://github.com/moment/luxon/issues/1497#issuecomment-1704350606

Most likely the backend JavaScript runtime is out of date.
